lib/task/import/validate/csvDigitalObjectUriValidator.class.php

```diff
@@ -0,0 +1,144 @@
+
+        if (!isset($this->digitalObjectUriColumnPresent)) {
+            $this->digitalObjectUriColumnPresent = array_key_exists('digitalObjectUri', $row);
+        }
+
+        if ($this->digitalObjectUriColumnPresent) {
+            if (!empty($row['digitalObjectUri'])) {
+                $this->addToUsageSummary($row['digitalObjectUri']);
+            }
+        }
+    }
+
+    public function getTestResult()
+    {
+        if (false == $this->digitalObjectUriColumnPresent) {
+            $this->testData->addResult(sprintf("Column 'digitalObjectUri' not present in CSV. Nothing to verify."));
+        } else {
+            $this->testData->addResult(sprintf("Column 'digitalObjectUri' found."));
+
+            if (empty($this->digitalObjectUses)) {
+                $this->testData->addResult(sprintf("Column 'digitalObjectUri' is empty."));
+            } else {
+                $digitalObjectUrisUsedMoreThanOnce = $this->getUsedMoreThanOnce();
+
+                if (!empty($digitalObjectUrisUsedMoreThanOnce)) {
+                    $this->testData->setStatusWarn();
+                    $this->testData->addResult(sprintf('Repeating Digital object URIs found in CSV.'));
+
+                    foreach ($digitalObjectUrisUsedMoreThanOnce as $uri) {
+                        $this->testData->addDetail(sprintf("Number of duplicates for URI '%s': %s", $uri, $this->digitalObjectUses[$uri]));
+                    }
+                }
+
+                $invalidUris = $this->getInvalidUris();
+
+                if (!empty($invalidUris)) {
+                    $this->testData->setStatusError();
+                    $this->testData->addResult(sprintf('Invalid digitalObjectUri values detected: %s', count($invalidUris)));
+
+                    foreach ($invalidUris as $file) {
+                        $this->testData->addDetail(sprintf('Invalid URI: %s', $file));
+                    }
+                }
+            }
+        }
+
+        return parent::getTestResult();
+    }
+
+    protected function getUsedMoreThanOnce()
+    {
+        $usedMoreThanOnce = [];
+
+        foreach ($this->digitalObjectUses as $digitalObjectUri => $uses) {
+            if ($uses > 1) {
+                array_push($usedMoreThanOnce, $digitalObjectUri);
+            }
+        }
+
+        return $usedMoreThanOnce;
+    }
+
+    protected function addToUsageSummary($value)
+    {
+        $this->digitalObjectUses[$value] = (!isset($this->digitalObjectUses[$value])) ? 1 : $this->digitalObjectUses[$value] + 1;
+    }
+
+    private function getInvalidUris()
+    {
+        $invalidUris = [];
+
+        foreach ($this->digitalObjectUses as $uri => $uses) {
+            if (false === filter_var($uri, FILTER_VALIDATE_URL)) {
+                array_push($invalidUris, $uri);
+            } else {
+                $parsedUri = parse_url($uri);
+
+                if (!('https' == $parsedUri['scheme'] || 'http' == $parsedUri['scheme'])) {
```

Comment:
Opinion: I find it easier to read conditionals when multilple terms aren't negated, e.g.

```
if ('https' != $parsedUri['scheme'] && 'http' != $parsedUri['scheme']) {
```

Reply:
👍